packages/flutter/lib/src/material/floating_action_button_location.dart

```diff
@@ -49,27 +73,157 @@ abstract class FloatingActionButtonLocation {
   /// const constructors so that they can be used in const expressions.
   const FloatingActionButtonLocation();
 
+  /// Start-aligned [FloatingActionButton], floating over the transition between
+  /// the [Scaffold.appBar] and the [Scaffold.body].
+  ///
+  /// To align a floating action button with [CircleAvatar]s in the
+  /// [ListTile.leading] slots of [ListTile]s in a [ListView] in the [Scaffold.body],
+  /// use [miniStartTop] and set [FloatingActionButton.mini] to true.
+  ///
+  /// This is unlikely to be a useful location for apps that lack a top [AppBar]
+  /// or that use a [SliverAppBar] in the scaffold body itself.
+  static const FloatingActionButtonLocation startTop = _StartTopFABLocation();
+
+  /// Start-aligned [FloatingActionButton], floating over the transition between
+  /// the [Scaffold.appBar] and the [Scaffold.body], optimized for mini floating
+  /// action buttons.
+  ///
+  /// This is intended to be used with [FloatingActionButton.mini] set to true,
+  /// so that the floating action button appears to align with [CircleAvatar]s
+  /// in the [ListTile.leading] slot of a [ListTile] in a [ListView] in the
+  /// [Scaffold.body].
+  ///
+  /// This is unlikely to be a useful location for apps that lack a top [AppBar]
+  /// or that use a [SliverAppBar] in the scaffold body itself.
+  static const FloatingActionButtonLocation miniStartTop = _MiniStartTopFABLocation();
```

Comment:
You mentioned in the description that `miniCenterTop` and `miniCenterDocked` are not defined since they are just the same as their non-mini versions.

I think it might be more clear if they were defined and just pointed to the same private class, to make the API more consistent. Just a suggestion.

Reply:
Done:
https://github.com/flutter/flutter/pull/51465/files#diff-a25ef574fd876f03d590951007e032d3R543
https://github.com/flutter/flutter/pull/51465/files#diff-a25ef574fd876f03d590951007e032d3R639